File: Hacktoberfest-2023/Programming/C++/SORT/SelectionSort.cpp
#include <iostream>

void selectionSort(int arr[], int n) {
    for (int i = 0; i < n - 1; ++i) {
        int minIdx = i;
        for (int j = i + 1; j < n; ++j) {
            if (arr[j] < arr[minIdx])
                minIdx = j;
        }
        std::swap(arr[i], arr[minIdx]);
    }
}

int main() {
    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr) / sizeof(arr[0]);

    std::cout << "Original array: ";
    for (int i = 0; i < n; ++i)
        std::cout << arr[i] << " ";
    std::cout << "\n";

    selectionSort(arr, n);

    std::cout << "Sorted array: ";
    for (int i = 0; i < n; ++i)
        std::cout << arr[i] << " ";
    std::cout << "\n";

    return 0;
}

File: Sorting_Algorithms/Selection Sort.py
# Function to implement Selection Sort

def selection_sort(arr):
    l = len(arr)
    
    # Iterate through the entire list.
    for i in range(l):

        for j in range(i + 1, l):
            
            # Check if the current element is greater than the element at the next position.            
            if arr[i] > arr[j]:
                # Swap the elements if they are out of order.
                arr[i], arr[j] = arr[j], arr[i]
    
    return arr



if __name__ == '__main__':
    List = [1, 5, 2, 90, 4, 67, 23, 12, 68]

    # Print the original list.
    print("Original List:")
    print(List)

    # Sort the list using the selection_sort function and print the sorted list.
    sorted_list = selection_sort(List)
    print("Sorted List:")
    print(sorted_list)
